### PR TITLE
core: upgrade ceph to latest release v17.2.5

### PR DESCRIPTION
**Pull request by subhamkrai**
Signed-off-by: subhamkrai <srai@redhat.com>

**Description of your changes:**
core: upgrade ceph to latest release v17.2.5

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: images/ceph/Makefile [hunk 1]
```diff
@@ -18,9 +18,9 @@ include ../image.mk
 # Image Build Options
 
 ifeq ($(GOARCH),amd64)
-CEPH_VERSION ?= v17.2.3-20220804
+CEPH_VERSION ?= v17.2.5-20221017
 else
-CEPH_VERSION ?= v17.2.3-20220805
+CEPH_VERSION ?= v17.2.5-20221017
```
Comment (subhamkrai) [inline images/ceph/Makefile:23]: @travisn where to ceph release for diff arch?

File: deploy/examples/images.txt [hunk 2]
```diff
@@ -1,5 +1,10 @@
+<<<<<<< Updated upstream
```
Comment (travisn) [inline deploy/examples/images.txt:1]: resolution needed

File: design/ceph/ceph-nfs-ganesha.md [hunk 3]
```diff
@@ -220,9 +220,9 @@ Directory, and FreeIPA.
 Prototype information detailed on Rook blog:
 https://blog.rook.io/prototyping-an-nfs-connection-to-ldap-using-sssd-7c27f624f1a4
 
-NFS-Ganesha (via libraries within its container) is the client to SSSD. As of Ceph v17.2.3, the Ceph
```
Comment (travisn) [inline design/ceph/ceph-nfs-ganesha.md:223]: Let's revert this file, it's telling a specific version where the fix and feature were available.

File: images/ceph/Makefile (hunk 1 above)
Comment (travisn) [reply images/ceph/Makefile:23]: This looks correct. See the arm64 image tags [here](https://quay.io/repository/ceph/ceph-arm64?tag=latest&tab=tags)

File: design/ceph/ceph-nfs-ganesha.md [hunk 4]
```diff
@@ -220,9 +220,9 @@ Directory, and FreeIPA.
...
+NFS-Ganesha (via libraries within its container) is the client to SSSD. As of Ceph v17.2.5, the Ceph
 container image does not have the `sssd-client` package installed which is required for supporting
-SSSD. It should be available in Ceph v17.2.4.
+SSSD. It should be available in Ceph v17.2.5.
```
Comment (travisn) [inline design/ceph/ceph-nfs-ganesha.md:225]: @BlaineEXE Can we change this to `is`?
```suggestion
SSSD. It is available starting from Ceph v17.2.4.
```

File: pkg/operator/ceph/nfs/security_test.go [hunk 5]
```diff
@@ -51,7 +51,7 @@ func mockReconcile() *ReconcileCephNFS {
 		},
 		cephClusterSpec: &cephv1.ClusterSpec{
 			CephVersion: cephv1.CephVersionSpec{
-				Image: "quay.io/ceph/ceph:v17.2.4",
```
Comment (travisn) [inline pkg/operator/ceph/nfs/security_test.go:54]: @BlaineEXE Does this test need the specific tag, or could we just use the `v17` tag so we don't update it with every ceph patch?

**Review by travisn — CHANGES_REQUESTED** [review]

File: design/ceph/ceph-nfs-ganesha.md (hunk 4 above)
Comment (BlaineEXE) [reply design/ceph/ceph-nfs-ganesha.md:225]: Yes, I think that change looks good.

File: pkg/operator/ceph/nfs/security_test.go (hunk 5 above)
Comment (BlaineEXE) [reply pkg/operator/ceph/nfs/security_test.go:54]: `v17` is fine

File: Documentation/Upgrade/ceph-upgrade.md [hunk 6]
```diff
@@ -41,7 +41,7 @@ please see the [v1.8 upgrade guide for a special upgrade consideration](https://
 
 !!! warning
     Ceph v17.2.2 has a blocking issue when running with Rook. If you are running Ceph v17, we
-    recommend using v17.2.3 or newer.
+    recommend using v17.2.5 or newer.
```
Comment (travisn) [inline Documentation/Upgrade/ceph-upgrade.md:44]: Let's revert this doc update

File: Documentation/CRDs/Object-Storage/ceph-object-store-crd.md [hunk 7]
```diff
@@ -242,7 +242,7 @@ vault write -f transit/keys/<mybucketkey> exportable=true # transit engine
 
 * `tokenSecretName` can be (and often will be) the same for both kms and s3 configurations.
 
-* `AWS-SSE:S3` requires Ceph Quincy (v17.2.3) and later.
+* `AWS-SSE:S3` requires Ceph Quincy (v17.2.5) and later.
```
Comment (travisn) [inline Documentation/CRDs/Object-Storage/ceph-object-store-crd.md:245]: Let's revert this, it should still be v17.2.3

**Review by travisn — CHANGES_REQUESTED** [review]

File: Documentation/Upgrade/ceph-upgrade.md (hunk 6 above)
Comment (subhamkrai) [reply Documentation/Upgrade/ceph-upgrade.md]: done

File: design/ceph/ceph-nfs-ganesha.md (hunk 4 above)
Comment (subhamkrai) [reply design/ceph/ceph-nfs-ganesha.md]: done

File: deploy/examples/images.txt (hunk 2 above)
Comment (subhamkrai) [reply deploy/examples/images.txt]: done

**Review by travisn — APPROVED** [review]

**Pull request merged**